### PR TITLE
Revises the overview section of the standards...

### PR DESCRIPTION
**Pull request by ekidd**
...as a precursor for the updates to come in v2.0.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,33 +14,22 @@
...
+* Consider code to be both an artifact that promotes adoption and suggests best practices in its composition.
 
-### Expectations
 
-* Understand and follow the standards outlined in this document.
-* Understand and follow the [methodology](/standards/methodology.md/).
-* Write code knowing that someone else might be maintaining it.
-* Select tools that benefit the project.
+### Standards
+The standards are hard rules which should be enforceable via automated tools like linters; they are the base upon which consistent and interchangeable code is written.
```
Comment (nategeslin) [inline README.md]: this sentence doesn't make sense or may just be missing a word.

File: README.md [hunk 2]
```diff
@@ -14,33 +14,22 @@
...
 
-For a successful project, always favor solutions that benefit the project and end users over producing "perfect code".
+### Guiding Principles
 
-Always aim to meet the standards knowing you'll have to break the rules at some point. Be thoughtful about when you deviate and check with a senior software engineer or a principal software engineer for guidance.
+* Code should first and foremost be resilient.
+* Solutions should focus on the best approach for a given problem space, and should not be limited by adherence to implementation details.
+* Discretion should be exercised when selecting tools to solve a problem, preferring those that inherently resonate with the methodology over those that do not.
+* The methodology should be agnostic to the development context in which it is being applied.
+* Consider best practices as thematic and conceptual guidelines, not literal dictates.
```
Comment (RyanNerdery) [inline README.md]: Suggest: not literal dictations*

File: README.md (hunk 1 above)
Comment (ekidd) [reply README.md]: It is a bit wordy - what specifically is throwing you off?

File: README.md [hunk 3]
```diff
@@ -14,33 +14,22 @@
...
+* Consider code to be both an artifact that promotes adoption and suggests best practices in its composition.
```
Comment (RyanNerdery) [inline README.md]: Suggest: Either simply remove "both" or move it.

Consider code to be an artifact that [both] promotes adoption and suggests best practices in its composition

File: README.md (hunk 1 above)
Comment (RyanNerdery) [reply README.md]: I think it makes sense.  The usage of the word "that" instead of "which" (hard rules that should be) sounds a little better to me, though, I think.

Comment (nategeslin) [reply README.md]: To be honest, I just had to read it a few more times.  working off of @RyanNerdery suggestion, I feel that changing `which` to `that` makes a difference in readability.

Comment (mliq) [reply README.md]: It does make sense to me, but do you think a form like this would be more clear @nategeslin? 

`The standards are hard rules: they are the base upon which consistent and interchangeable code is written, and should be enforceable via automated tools (linters).`

Comment (ekidd) [reply README.md]: Hm, it certainly seems some grammar clarification is in order here. We'll update based on some of the suggestions here.

Comment (nategeslin) [reply README.md]: @mliq words are hard.  I do think that reads better!

File: README.md [hunk 4]
```diff
@@ -14,33 +14,22 @@
...
 
-The standards herein are a guide to keep developers at the Nerdery interoperable amongst each other and able to produce the highest quality code given the timeline and budget constraints of a project.
+This is a living document and is reflective of the opinions and expertise of The Nerdery’s front-end engineering group. However, the discussion of, contribution to, and evolution of this document is reliant on the front-end community both inside and outside of The Nerdery.
 
-For a successful project, always favor solutions that benefit the project and end users over producing "perfect code".
```
Comment (thorn100) [inline README.md]: I think the point that we first and foremost have the project and users in mind when writing our code versus creating "perfect code" is a really important point to make. Creating readable and reusable code over perfect or clever code should be preferred.

File: README.md [hunk 5]
```diff
@@ -14,33 +14,22 @@
...
+### Guiding Principles
```
Comment (kevtronkrueger) [inline README.md]: Are these intended to be guiding principles of a methodology, or the guiding principles for a developer operating within that methodology? I think there's a bit of an identity conflict here.

> ...preferring those that inherently resonate with the methodology over those that do not

seems to imply that we're talking about how to do something within a preexisting methodology, while

> The methodology should be agnostic to the development context in which it is being applied

speaks in a way that says the "methodology" is something that's actively being created.

File: README.md [hunk 6]
```diff
@@ -14,33 +14,22 @@
...
 
-Always aim to meet the standards knowing you'll have to break the rules at some point. Be thoughtful about when you deviate and check with a senior software engineer or a principal software engineer for guidance.
```
Comment (thorn100) [inline README.md:23]: Noting when it's appropriate to deviate from the standards and when it's not emphasizes the point that all projects are different and exceptions may need to be made in the interest of the project's success. It's probably worth considering if there would there ever be a need to document those reasons for deviating from the standards thereby providing context for others on the project.

File: README.md (hunk 3 above)
Comment (kevtronkrueger) [reply README.md]: This whole line is suffering from too much jargon, and, in general, sounding a bit esoteric. I've read it 20 times and I'm still having trouble deriving the intent.

File: README.md [hunk 7]
```diff
@@ -14,33 +14,22 @@
...
+* Code should first and foremost be resilient.
+* Solutions should focus on the best approach for a given problem space, and should not be limited by adherence to implementation details.
+* Discretion should be exercised when selecting tools to solve a problem, preferring those that inherently resonate with the methodology over those that do not.
```
Comment (kevtronkrueger) [inline README.md]: The term "the methodology" could benefit from added clarity. Perhaps, "this methodology" or "Nerdery Front-End methodology"?

Comment (ekidd) [reply README.md]: `Methodology` is briefly defined further down the document, but will have it's own introductory statement in the Methodology section in a future PR. Would linking to that page the first time we mention it provide enough clarity?

File: README.md (hunk 6 above)
Comment (ekidd) [reply README.md:23]: I agree, it is important to understand that there is going to be times when deviation is necessary. With any given project having the potential to be radically different from another, mandating how and when deviations occur can be very constraining. That thought was the inspiration for some of the proposed changes:

> Because the front-end engineering ecosystem is constantly evolving, a set of guidelines for working within it must continue to be relevant even as specific implementations change.

> Solutions should focus on the best approach for a given problem space, and should not be limited by adherence to implementation details.

> Consider best practices as thematic and conceptual guidelines, not literal dictates.

The thought is that focusing on building the right solution for a problem and having guidelines which ensure consistency regardless of implementation would reduce the need to "break the rules". The guidelines inherently expect that implementations will need to differ from project to project.

File: README.md (hunk 3 above)
Comment (ekidd) [reply README.md]: Hm, that's a fair analysis. What conclusion did you reach? Knowing what you got out of it can help show what needs clarification.

File: README.md (hunk 6 above)
Comment (thorn100) [reply README.md:23]: Perhaps I'm just getting lost in the language of it then. Ideally, I'd prefer the points you highlighted above were more plainly stated. Something like...

> Solutions should focus on the best approach for a given problem or project. They should not be limited by a strict interpretation of these standards.

> Consider these standards as "best practices"...

My 2¢. Thanks for humoring me. 

Regarding the second part of my point... when a project does need to deviate from the standards, do you think it would be worthwhile documenting why and specifically how any deviations occurred? I'd think this would also help to encourage developers to justify their reasoning as well as provide contextual information for other developers (current and future) on the project.

Comment (ekidd) [reply README.md:23]: Personally, yes, I believe documenting deviations are fruitful. I also believe it to be a best practice, something that will take many different forms depending on the needs of the particular problem. There is an opportunity to expand on what deviation from the standards/methodology means in the methodology section - do you feel this would be a more beneficial place to express that perspective?

File: README.md (hunk 4 above)
Comment (ekidd) [reply README.md]: This point was actually one which we really wanted to expand upon. The exact wording has changed, but the idea you highlighted is a theme we tried to seed throughout the proposed changes:

> The Nerdery’s perspective is that code exists to solve a problem.

> We favor solutions which are pragmatic and innovative...

> Code should first and foremost be resilient.

> We prioritize interoperability when choosing solutions, and use our guiding principles to write code that is resilient and easy to maintain.

File: README.md (hunk 1 above)
Comment (ekidd) [reply README.md]: This sentence has been revised in 59c6f1b.

File: README.md (hunk 5 above)
Comment (ekidd) [reply README.md]: The guiding principles have been updated to share a voice and be entirely consumer centric (not a guidepost for the updates being made themselves). this is reflected in 59c6f1b.

File: README.md (hunk 3 above)
Comment (ekidd) [reply README.md]: This guiding principle has been updated to be simpler and more specific for 59c6f1b. It also is now listed second, not last.

File: README.md (hunk 2 above)
Comment (ekidd) [reply README.md]: Wording for this principle was updated in 59c6f1b.

File: README.md (hunk 6 above)
Comment (ekidd) [reply README.md:23]: The guiding principle you mentioned has been updated in 59c6f1b. It now much more plainly states it's intent.

File: README.md (hunk 7 above)
Comment (ekidd) [reply README.md]: The methodology section is now linked to at it's first mention in 59c6f1b.

**Review by bbellx — APPROVED** [review]
Thank you all for your input! This is an excellent kickoff to what will be an exciting and important update to our front-end standards. You're continued involvement is much appreciated.
